snuba/subscriptions/codecs.py

```diff
@@ -10,29 +10,29 @@
 from snuba.query.exceptions import InvalidQueryException
 from snuba.subscriptions.data import (
     ScheduledSubscriptionTask,
+    SnQLSubscriptionData,
     Subscription,
-    SubscriptionData,
     SubscriptionIdentifier,
     SubscriptionTaskResult,
     SubscriptionWithMetadata,
 )
 from snuba.utils.codecs import Codec, Encoder
 
 
-class SubscriptionDataCodec(Codec[bytes, SubscriptionData]):
+class SubscriptionDataCodec(Codec[bytes, SnQLSubscriptionData]):
```

Comment:
Shouldn't this still take a `SubscriptionData` instead of `SnQLSubscriptionData`? Once you add the RPC type, that's what will need to happen here.

Reply:
Yes, you're right! I was doing union types for some reason when I introduce the RPC class, this absolutely did not cross my mind 😅

Reply:
addressed in https://github.com/getsentry/snuba/pull/6419/commits/97a0360ffed6c0c10981c776611eeafea55147f1